src/sentry/integrations/perforce/client.py

```diff
@@ -29,42 +33,97 @@ class PerforceClient(RepositoryClient, CommitContextClient):
+        self.org_integration = org_integration
         self.P4 = P4
         self.P4Exception = P4Exception
 
+        # Extract configuration from org_integration
+        if not org_integration:
+            raise IntegrationError("Organization Integration is required for Perforce")
+
+        config = org_integration.config
+        self.p4port = config.get("p4port", "localhost:1666")
+        self.user = config.get("user", "")
+        self.password = config.get("password")
+        self.client_name = config.get("client")
+        self.ssl_fingerprint = config.get("ssl_fingerprint")
+
     def _connect(self):
-        """Create and connect a P4 instance with SSL support."""
-        pass
+        """
+        Create and connect a P4 instance with SSL support.
+
+        Uses P4Python API:
+        - p4.connect(): https://www.perforce.com/manuals/p4python/Content/P4Python/python.programming.html#python.programming.connecting
+        - p4.run_trust(): https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_trust.html
+        - p4.run_login(): https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_login.html
+        """
+        p4 = self.P4()
+        p4.port = self.p4port
+        p4.user = self.user
+        p4.password = self.password
+
+        if self.client_name:
+            p4.client = self.client_name
+
+        p4.exception_level = 1  # Only errors raise exceptions
+
+        try:
+            p4.connect()
+
+            if self.ssl_fingerprint and self.p4port.startswith("ssl:"):
+                try:
+                    p4.run_trust("-i", self.ssl_fingerprint)
+                except self.P4Exception as trust_error:
+                    p4.disconnect()
+                    raise ApiError(
+                        f"Failed to establish SSL trust: {trust_error}. "
+                        f"Ensure ssl_fingerprint is correct. Obtain with: p4 -p {self.p4port} trust -y"
+                    )
+
+            if self.password:
+                try:
+                    p4.run_login()
+                except self.P4Exception as login_error:
+                    p4.disconnect()
+                    raise ApiError(f"Failed to authenticate with Perforce: {login_error}")
+
+            return p4
+        except self.P4Exception as e:
+            error_msg = str(e)
+            # Provide helpful error message for SSL issues
+            if "SSL" in error_msg or "trust" in error_msg.lower():
+                raise ApiError(
+                    f"Failed to connect to Perforce (SSL issue): {error_msg}. "
+                    f"Ensure ssl_fingerprint is correct. Obtain with: p4 -p {self.p4port} trust -y"
+                )
+            raise ApiError(f"Failed to connect to Perforce: {error_msg}")
 
     def _disconnect(self, p4):
         """Disconnect P4 instance."""
-        pass
+        try:
+            if p4.connected():
+                p4.disconnect()
+        except Exception:
+            pass
```

Comment:
if we fail to disconnect, would there be any lingering effects we'd be interested to log ?